pennylane/compiler/__init__.py

```diff
@@ -170,9 +179,10 @@ def circuit(x: int):
 Adding a compiler
 -----------------
 
-For any compiler packages seeking to be registered, it is imperative that they
-expose the ``entry_points`` metadata under the designated group name
-``pennylane.compilers``, with the following entry points:
+To register any compiler packages, an experimental interface is available,
```

Comment:
i would make it extra-obvious, just in case. like in its own paragraph before, put something like:

> 🚨 The PennyLane compiler API is experimental and subject to change 🚨

Reply:
Let's make this a warning directive.